server/backends/blobstore/aws/aws.go

```diff
@@ -254,3 +255,23 @@ func (a *AwsS3BlobStore) BlobExists(ctx context.Context, blobName string) (bool,
 
 	return true, nil
 }
+
+func (a *AwsS3BlobStore) Writer(ctx context.Context, blobName string) (io.WriteCloser, error) {
+	// Open a pipe.
+	pr, pw := io.Pipe()
+
+	errch := make(chan error)
+
+	// Upload from pr in a separate Go routine.
+	go func() {
```

Comment:
In these async goroutine funcs, we should begin the goroutine with `defer pr.Close()` as the first line, so that if the async blobstore upload fails, any subsequent writes to the returned CompressedBlobstoreWriter would fail (due to the pipe being closed), rather than getting stuck.

Example: https://go.dev/play/p/2TQ4JP8y10w